lib/foods.js

```diff
@@ -34,39 +42,45 @@ function createDeleteButton() {
   });
 };
 
-function saveFoodList(whichTable){
-  var foods = [];
-  whichTable.children().children('.food-row').each(function(){
-    var name = $(this).children('.food-name').text();
-    var calories = $(this).children('.food-calories').text();
-    foods.push(createNewFood(name, calories));
-  });
-  localStorage.setItem(storedTableName(whichTable), JSON.stringify(foods));
-  return foods;
+function saveFoodList(location, data){
+  localStorage.setItem(location, JSON.stringify(data));
+  return data;
 };
 
-function storedTableName(whichTable){
-  if (whichTable.attr('id') == "food-list") {
-    return "foods";
-  } else if (whichTable.attr('id') == "meal-breakfast") {
-    return "mealBreakfast";
-  };
-};
+function updateFoodList(location, updatedObject){
+  // get appropriate food list, make into array
+
+  var foodsRaw = localStorage.getItem(location);
+  var data = $.parseJSON(foodsRaw);
+
+  // match = $.grep(data, function(e){ return e.id == updatedObject.id });
+  // filter out object
+  data = data.filter(function(element) { return element.id != updatedObject.id;});
+  // push in updated object
+  data.push(updatedObject);
+  saveFoodList(location, data);
+}
 
 function updateLineListener() {
-  $('td').on('click', function() {
-    var $this = $(this);
+  $('#food-list .food-row').bind('click', function(event) {
+    var $this = $(event.target);
+    var $row = $(this);
     var $input = $('<input>', {
       value: $this.text(),
       type: 'text',
       blur: function() {
         $this.text(this.value);
-        saveFoodList($('#food-list'));
+        var objectId = $this.parent().children('.hidden').text();
```

Comment:
@lucyconklin Broke out this long function here into a separate function. Included in PR #11